workspaces/rbac/plugins/rbac-backend/src/providers/connect-providers.ts

```diff
@@ -181,39 +184,34 @@ export class Connection implements RBACProviderConnection {
         }
 
         const singleRole = roleMeta && currentRole.length === 1;
-
-        let eventName: string;
-        let message: string;
+        let eventName: (typeof RoleEvents)[keyof typeof RoleEvents] =
```

Comment:
```
let eventName: (typeof RoleEvents)[keyof typeof RoleEvents] =   RoleEvents.ROLE_UPDATE; 
if (singleRole) {
   eventName = RoleEvents.ROLE_DELETE;
}
```
it can be simplified to:
```     
// Only one role exists in rbac remove role metadata as well
let eventName = singleRole ? RoleEvents.ROLE_DELETE : RoleEvents.ROLE_UPDATE;
```
But it is minor and up to you.